csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -240,18 +240,11 @@ class PredicateChcker : public IterVisitor {
         predicateNonDivisibleSplit(expr) || predicateExpandReduce(expr) ||
         predicateRNGOp(expr);
 
-    // A cp.async op would need a predicate for either the global
-    //  input or its shared mem output, or both.
-    // Due to the WAR discussed in [Predicate Inversion for CpAsync],
-    //  we currently cannot support use cases where both the gmem read
-    //  and the smem write need to be predicated.
-    // Adding a check here would make the exclusion of such case as precise as
-    //  possible and avoid duplication of predicateSharedMemAccess
-    //  logic. But this part along with [Predicate Inversion for CpAsync]
-    //  should be cleaned up all together when we extend predicate/masking
-    //  logic to cover this usage.
+    // For CpAsyncOp in matmul, must use inline predicate without IfThenElse
+    // predicate to ensure the out of bounday shared memory data is set to zero.
     NVF_ERROR(
-        !(ir_utils::isCpAsyncOp(expr) && needs_predicate_smem_access),
```

Comment:
Why not do something like `if (needs_predicate_smem_access) { don't use inline predicate }`?

Reply:
This logic is added in `UnrollPass`, where we won't use inline predicate if `needs_predicate_smem_access` is true.
So we can remove this `NVF_ERROR` check given matmul always uses static bdimx, bdimy, and bdimz (is this true?).